### PR TITLE
Small fix for value (notification) events

### PR DESCRIPTION
**Pull request by marcelveldt**
Value notification value is already provided in the event.
This change will make sure that the message signature is always consistent for all value events, where the value attribute always contains the most recent value.

### CODE REVIEW

File: src/lib/state.ts
Comment (balloob) [inline src/lib/state.ts:54]: If this is only used in the forwarder and it depends on the `valueArgs`, there is no need to extract it in a function.

File: src/lib/state.ts
Comment (balloob) [inline src/lib/state.ts:60]: If there is no need, let's not include it?

**Comment by marcelveldt** [comment]
I've simplified it a bit as it makes sense that the ValueState object is more generic.
The value key always contains the right value now. This is also easier in the python lib.
Strictly taken this should done with if-else (to account for value 0) but this looks cleaner imo and the getValue call is light.

File: src/lib/state.ts
Comment (marcelveldt) [reply src/lib/state.ts]: I wanted the object to have the same signature as ValueState. I've now changed it so it still uses the same dumpValue function.

File: src/lib/state.ts
Comment (marcelveldt) [reply src/lib/state.ts]: Decided to keep it in for consistency reasons.

File: src/lib/state.ts
Comment (balloob) [inline src/lib/state.ts:40]: why is this necessary and can't we put that in the function signature?

File: src/lib/forward.ts
Comment (balloob) [inline src/lib/forward.ts:128]: This is now not a change?

File: src/lib/state.ts
Comment (marcelveldt) [reply src/lib/state.ts:40]: I want to make sure that all value related messages generate the same valueState message.
So, the value can come from an event but also from enumerating the valueID's on the node.
It's either duplicating code or have one place that contains this logic.

File: src/lib/state.ts
Comment (balloob) [inline src/lib/state.ts:39]: Did you test this? It seems like we're changing the signature and miss out on typing ?

File: src/lib/forward.ts
Comment (marcelveldt) [reply src/lib/forward.ts:128]: No, with the consolidating the dumpValue this can all be handled

File: src/lib/state.ts
Comment (marcelveldt) [reply src/lib/state.ts:39]: Yes, I did test this. valueArgs can be either ValueChangedArgs, ValueAddedArgs etc. etc so it was either a lot of OR's types or abusing any.

File: src/lib/state.ts
Comment (marcelveldt) [reply src/lib/state.ts:40]: FYI: we add the ccVersion and metadata to all value messages.

Comment (marcelveldt) [reply src/lib/state.ts:40]: But if you want I could also just do the specifics in each function and use the dumpValue only for the dumpState

File: src/lib/state.ts
Comment (marcelveldt) [reply src/lib/state.ts]: adjusted

File: src/lib/state.ts
Comment (balloob) [reply src/lib/state.ts:44]: I'm just scared that we accidentally break something and typing won't find it. Is this line still necessary even with the translated value id type?

Comment (marcelveldt) [reply src/lib/state.ts:44]: Yes because we “convert” the incoming ValueID to ValueState, which includes value, metadata and ccVersion.

this line was already here btw, before this change.

File: src/lib/state.ts
Comment (marcelveldt) [inline src/lib/state.ts:69]: small typo, should be !"value" in valueArgs. Only if there is value in the valueArgs we retrieve it with node.getValue

Comment (marcelveldt) [reply src/lib/state.ts]: adjusted

**Comment by marcelveldt** [comment]
tested this branch on my test mesh. Works great.
Waiting on final approval from @balloob before merge.

**Review by balloob — APPROVED** [review]

**Comment by marcelveldt** [comment]
Changes made like discussed on discord.
The events "value_updated" and "value_removed" will be forwarded as-is. "value_added" and "value_notification" will be extended with metadata and ccVersion, same for a value that is dumped in the dumpNode function.

**Review by balloob — APPROVED** [review]

**Pull request merged**
